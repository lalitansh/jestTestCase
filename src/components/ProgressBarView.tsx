import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {Dialog} from 'react-native-simple-dialogs';
import style from '../constants/theme/Style';
import {color} from '../constants/theme/Color';
import {MaterialIndicator} from 'react-native-indicators';

type Props = {
  visible: boolean;
  loadingText?: string;
};

type State = {};

export default class ProgressBarView extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
  }

  render() {
    const {loadingText, visible} = this.props || {};
    if (visible) {
      return (
        <Dialog dialogStyle={style.progressDialogBox} visible={true}>
          <View style={styles.viewStyle}>
            <MaterialIndicator
              color={color.primary}
              size={40}
              animationDuration={5200}
              trackWidth={3}
            />

            {loadingText && (
              <Text style={{marginTop: 5, textAlign: 'center'}}>
                {loadingText}
              </Text>
            )}
          </View>
        </Dialog>
      );
    } else {
      return <View />;
    }
  }
}

const styles = StyleSheet.create({
  viewStyle: {
    height: 40,
    alignItem: 'center',
  },
});
