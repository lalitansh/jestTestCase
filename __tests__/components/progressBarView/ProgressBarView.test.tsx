import { shallow } from 'enzyme';
import React from 'react';
import ProgressBarView from '../../../src/components/ProgressBarView';

const mockVisible = {
  visible: false,
};

describe('LoadingIndicator', () => {
  const container = shallow(<ProgressBarView {...mockVisible} />)
  describe('Rendering', () => {
    it('should match to snapshot', () => {

      expect(container).toMatchSnapshot()
    });

    it('should have an Dialog component', () => {
      expect(container.find('Dialog[visible=true]'));
    });

    it('should have an View component', () => {
      expect(container.find('View'));
    });

    it('should have an MaterialIndicator component', () => {
      expect(container.find('MaterialIndicator'));
    });
  });
});
