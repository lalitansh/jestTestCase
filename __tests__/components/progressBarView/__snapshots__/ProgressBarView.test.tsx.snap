// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
